### PR TITLE
[PW-1535]Add CheckoutPaymentAction

### PR DESCRIPTION
**Pull request by AlexandrosMor**

### CODE REVIEW

**Comment by AlexandrosMor** [comment]
closes #115 
closes #126

File: Adyen/Model/Checkout/CheckoutPaymentsAction.cs [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+            /// </summary>
+            [EnumMember(Value = "threeDS2Fingerprint")]
+            ThreeDS2Fingerprint = 1,
+            /// <summary>
+            /// Enum voucher for value: voucher
+            /// </summary>
+            [EnumMember(Value = "voucher")]
+            Voucher = 2,
+            /// <summary>
+            /// Enum qrCode for value: qrCode
```
Comment (Aleffio) [inline Adyen/Model/Checkout/CheckoutPaymentsAction.cs]: minor
```suggestion
            /// Enum redirect for value: redirect
```

File: Adyen/Model/Checkout/CheckoutPaymentsAction.cs [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+        }
+
+        /// <summary>
+        /// Returns true if objects are equal
+        /// </summary>
+        /// <param name="input">Object to be compared</param>
+        /// <returns>Boolean</returns>
+        public override bool Equals(object input)
+        {
+            return this.Equals(input as PaymentsResponse);
```
Comment (Aleffio) [inline Adyen/Model/Checkout/CheckoutPaymentsAction.cs]:
```suggestion
            return this.Equals(input as CheckoutPaymentsAction);
```

File: Adyen/Model/Checkout/CheckoutPaymentsAction.cs [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+        }
+
+        public IEnumerable<ValidationResult> Validate(ValidationContext validationContext)
+        {
+            yield break;
+        }
+
+
+        /// <summary>
+        /// Returns true if PaymentResponse instances are equal
```
Comment (Aleffio) [inline Adyen/Model/Checkout/CheckoutPaymentsAction.cs]:
```suggestion
        /// Returns true if CheckoutPaymentsAction instances are equal
```

File: Adyen/Model/Checkout/CheckoutPaymentsAction.cs [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+        /// </summary>
+        /// <param name="input">Instance of PaymentResponse to be compared</param>
```
Comment (Aleffio) [inline Adyen/Model/Checkout/CheckoutPaymentsAction.cs]:
```suggestion
        /// <param name="input">Instance of CheckoutPaymentsAction to be compared</param>
```

File: Adyen/Model/Checkout/PaymentsResponse.cs [hunk 5]
```diff
@@ -95,7 +95,7 @@ public enum ResultCodeEnum
...
...ed for 3D Secure 2 transactions. * **Error** – There was an error when the payment was being processed. The reason is given in the &#x60;refusalReason&#x60; field. This is a final state. * **IdentifyShopper** – The issuer requires the shopper&#x27;s device fingerprint before the payment can be authenticated. Returned for 3D Secure 2 transactions. * **Refused** – Indicates the payment was refused. The reason is given in the &#x60;refusalReason&#x60; field. This is a final state. * **Pending** – Indicates that it is not possible to obtain the final status of the payment. This can happen if the systems providing final status information for the payment are unavailable, or if the shopper needs to take further action to complete the payment. For more information on handling a pending payment, refer to [Payments with pending status](https://docs.adyen.com/development-resources/payments-with-pending-status). * **Received** – Indicates the payment has successfully been received by Adyen, and will be processed. This is the initial state for all payments. * **RedirectShopper** – Indicates the shopper should be redirected to an external web page or app to complete the authorisation. .</param>
```
Comment (Aleffio) [inline Adyen/Model/Checkout/PaymentsResponse.cs]: not sure if the description of the Action is required

**Pull request merged**